[WIP] Dubovik Pavel

dubovikmaster opened the pull request:
my first commit

dubovikmaster committed:
initial commit

my first commit

dubovikmaster committed:
add setup

dubovikmaster committed:
add byild_parse function

dubovikmaster committed:
move file in folder

dubovikmaster committed:
rename and syntax error fix

dubovikmaster committed:
Update myfunction.py

dubovikmaster committed:
rename module

dubovikmaster committed:
Merge branch 'master' of https://github.com/dubovikmaster/PythonHomework

dubovikmaster committed:
rename module

dubovikmaster committed:
rename file

dubovikmaster committed:
edit dict function and syntax fix

dubovikmaster committed:
pep8 syntax edit

dubovikmaster committed:
setup: add find_packages

dubovikmaster committed:
pep8 syntax edit

dubovikmaster committed:
add folder mycalc

AlexeiBuzuma commented on final_task/mycalc/mymodule.py:
```diff
@@ -0,0 +1,160 @@
+import math
+import operator
+
+A = {'*': operator.mul, '%': operator.mod, '//': operator.floordiv, '^': operator.pow,
```
Имя переменной из одного символа. Не понятно, что в ней хранится.

dubovikmaster replied:
Исправил

AlexeiBuzuma commented on final_task/mycalc/mymodule.py:
```diff
@@ -0,0 +1,160 @@
+
+
+def first_foo(stroka):
+    stroka = stroka.strip()
```
транслит

dubovikmaster replied:
Исправил

AlexeiBuzuma commented on final_task/mycalc/mymodule.py:
```diff
@@ -0,0 +1,160 @@
+
+
+def find_comparsion(stroka):
+    i1 = 0
```
имя переменной не обозначает, что в ней хранится.

dubovikmaster replied:
Исправил

AlexeiBuzuma commented on final_task/mycalc/mymodule.py:
```diff
@@ -0,0 +1,160 @@
+    i1 = 0
+    lst = []
+    op = []
+    tup = ('>', '<', '!', '=')
```
имя переменной не обозначает, что в ней хранится.

dubovikmaster replied:
Исправил

AlexeiBuzuma commented on final_task/mycalc/mymodule.py:
```diff
@@ -0,0 +1,160 @@
+
+def find_comparsion(stroka):
+    i1 = 0
+    lst = []
```
имя переменной не обозначает, что в ней хранится.

dubovikmaster replied:
Исправил

AlexeiBuzuma commented on final_task/mycalc/mymodule.py:
```diff
@@ -0,0 +1,160 @@
+            op.append(elem)
+            i1 = i2 + 1
+    lst.append(stroka[i1:])
+    col = lst.count('')
```
имя переменной не обозначает, что в ней хранится.
такая проблема есть в большом количестве мест программы

dubovikmaster replied:
Исправил

AlexeiBuzuma commented on final_task/mycalc/mymodule.py:
```diff
@@ -0,0 +1,160 @@
+
+
+def find_brackets(stroka):
+    global indx, valbr
```
Зачем тут нужны глобальные перменные?

dubovikmaster replied:
Исправил

dubovikmaster commented:
исправлю все. Еще не занимался вылизыванием синтаксиса

пт, 19 апр. 2019 г. в 17:20, Aliaksei Buziuma <notifications@github.com>:

> *@AlexeiBuzuma* commented on this pull request.
> ------------------------------
>
> In final_task/mycalc/mymodule.py
> <https://github.com/PythonAndGoCourses2/PythonHomework/pull/12#discussion_r276998391>
> :
>
> > +    res = s[0]
>
> +    for idx, elem in enumerate(o):
>
> +        res = A[elem](res, s[idx+1])
>
> +    return res
>
> +
>
> +
>
> +def calc(stroka):
>
> +    stroka = '0+' + stroka
>
> +    stroka = del_space(stroka)
>
> +    stroka = replace_many_plus_minus(stroka)
>
> +    lst = plus_reject(stroka)
>
> +    return sum(list(map(result, lst)))
>
> +
>
> +
>
> +def find_brackets(stroka):
>
> +    global indx, valbr
>
>
> Зачем тут нужны глобальные перменные?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/PythonAndGoCourses2/PythonHomework/pull/12#pullrequestreview-228707332>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALVWLFQPRU6FPFGXN5VCIBTPRHILJANCNFSM4HG7ELNQ>
> .
>

AlexeiBuzuma commented on final_task/mycalc/pycalc.py:
```diff
@@ -0,0 +1,40 @@
+from mycalc import mymodule
+
+
+parser = argparse.ArgumentParser(description='Pure-python command line calculator.')
```
Я очень не рекомендую писать код напрямую в глобальной секции модуля. 
Вся логика должна находиться в функциях/методах. В глобальной области должны находится только определение функций, классов, констант и импорт модулей

dubovikmaster replied:
Исправил

dubovikmaster commented:
ты имеешь ввиду создание парсера запихнуть в функцию?


пт, 19 апр. 2019 г. в 17:24, Aliaksei Buziuma <notifications@github.com>:

> *@AlexeiBuzuma* commented on this pull request.
> ------------------------------
>
> In final_task/mycalc/pycalc.py
> <https://github.com/PythonAndGoCourses2/PythonHomework/pull/12#discussion_r276999272>
> :
>
> > @@ -0,0 +1,40 @@
>
> +
>
> +import argparse
>
> +from mycalc import mymodule
>
> +
>
> +
>
> +parser = argparse.ArgumentParser(description='Pure-python command line calculator.')
>
>
> Я очень не рекомендую писать код напрямую в глобальной секции модуля.
> Вся логика должна находиться в функциях/методах. В глобальной области
> должны находится только определение функций, классов, констант и импорт
> модулей
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/PythonAndGoCourses2/PythonHomework/pull/12#pullrequestreview-228708489>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALVWLFWRRBT72UQI6DHZ3ZTPRHIYVANCNFSM4HG7ELNQ>
> .
>

AlexeiBuzuma commented on final_task/mycalc/pycalc.py:
```diff
@@ -0,0 +1,40 @@
+    x, y = e.args
+    print(x, y)
+except ValueError as e:
+        print('error: invalid syntax', str(e), sep=' ')
```
табуляция уехала))

dubovikmaster replied:
Исправил

AlexeiBuzuma commented on final_task/mycalc/pycalc.py:
```diff
@@ -0,0 +1,40 @@
+    print('error: overflow')
+
+
+
```
много лишних пустых строк, должна быть ровно одна

dubovikmaster replied:
Исправил

AlexeiBuzuma commented on final_task/pycalc.py:
```diff
@@ -0,0 +1,40 @@
+    a = mymodule.first_foo(a)
+    result1 = mymodule.find_brackets(a)
+    [compare, result2] = mymodule.find_comparsion(result1)
+    if len(compare) != 0:
```
```python
if compare:
   ...
```

dubovikmaster replied:
Исправил

AlexeiBuzuma commented on final_task/mycalc/mymodule.py:
```diff
@@ -0,0 +1,160 @@
+    return lst
+
+
+def result(stroka):
```
Название функции должно описывать какое-то действие (глагол)

dubovikmaster replied:
Исправил

dubovikmaster committed:
power bag fix and error correct

dubovikmaster committed:
syntax fix

dubovikmaster committed:
bag fix

dubovikmaster committed:
add mycalc.pycalc

dubovikmaster committed:
add import mymodule

dubovikmaster committed:
add main function

dubovikmaster committed:
pep8 syntax

dubovikmaster committed:
Alex's recommendation fix

Syntax fixes.
Added function calc_logical.
Fixed error calculating boolean values. Extended unit tests

dubovikmaster committed:
pep8 syntax fix

Syntax fix

dubovikmaster committed:
pep8 syntax fix

dubovikmaster committed:
many changes

removed translit. Added two modules that allow finding roots of polynomials of degree at most 4
Extended help in the settings of the argparse module

dubovikmaster committed:
pep8 syntax edit

dubovikmaster committed:
fix unittest bug and pep8 syntax

dubovikmaster committed:
pep8 syntax edit

it's real funy

dubovikmaster committed:
Added support for numbers in exponential form and much more

Added support for numbers in exponential form. Added an iterative method for solving approximate equations with real coefficients. Changed the logic of some functions. Extended test

dubovikmaster committed:
pep8 syntax fix

dubovikmaster committed:
error edit bug fix

AlexeiBuzuma commented on final_task/mycalc/equations.py:
```diff
@@ -0,0 +1,126 @@
+                  4: solve_polynom.solv_cubic_equation, 5: solve_polynom.solv_four_degree}
+
+
+def coefficient_expression(list_expression):
```
Было бы намного лучше, если была бы документация к функциям :)

dubovikmaster commented:
Леша, уехал в командировку за границу, совсем небыло времени закончить((
Виноват. Косяков осталось много

ср, 12 июня 2019 г. в 19:11, Aliaksei Buziuma <notifications@github.com>:

> *@AlexeiBuzuma* commented on this pull request.
> ------------------------------
>
> In final_task/mycalc/equations.py
> <https://github.com/PythonAndGoCourses2/PythonHomework/pull/12#discussion_r293000172>
> :
>
> > @@ -0,0 +1,126 @@
>
> +from mycalc import mymodule
>
> +from mycalc import solve_polynom
>
> +
>
> +
>
> +SOLVE_FUNCTION = {0: solve_polynom.solve_inf, 1: solve_polynom.solve_number,
>
> +                  2: solve_polynom.solv_linear_equation, 3: solve_polynom.solv_quartic_equation,
>
> +                  4: solve_polynom.solv_cubic_equation, 5: solve_polynom.solv_four_degree}
>
> +
>
> +
>
> +def coefficient_expression(list_expression):
>
>
> Было бы намного лучше, если была бы документация к функциям :)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/PythonAndGoCourses2/PythonHomework/pull/12?email_source=notifications&email_token=ALVWLFWXX2RSA53DVQN7X43P2EN45A5CNFSM4HG7ELN2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB3KXFJA#pullrequestreview-248869540>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALVWLFUJPXBI7HUW2EKLDA3P2EN45ANCNFSM4HG7ELNQ>
> .
>

AlexeiBuzuma commented:
Очень круто, что ты разбил все решение на достаточно маленькие функции. Большинство этих функций достаточно легко читаются. Было бы еще лучше, если бы у функций была документация :)
